paper/paper.md

```diff
@@ -0,0 +1,204 @@
+    for (ddc::DiscreteElement<Dim2> idx2 : slice.domain()) {
+        sum += slice(idx2);
+    }
+    return sum;
+}
+
+int main()
+{
+    // Initialize Kokkos runtime (automatically finalized at the end of scope)
+    Kokkos::ScopeGuard const kokkos_scope;
+
+    // Initialize DDC runtime (automatically finalized at the end of scope)
+    ddc::ScopeGuard const ddc_scope;
+
+    // Create a discrete domain for Dim1 with 5 elements
+    ddc::DiscreteDomain<Dim1> const dom1
+            = ddc::init_trivial_space<Dim1>(ddc::DiscreteVector<Dim1>(5));
+
+    // Create a discrete domain for Dim2 with 7 elements
+    ddc::DiscreteDomain<Dim2> const dom2
+            = ddc::init_trivial_space<Dim2>(ddc::DiscreteVector<Dim2>(7));
+
+    // Define a 2D discrete domain combining Dim1 and Dim2
+    ddc::DiscreteDomain<Dim1, Dim2> const dom(dom1, dom2);
+
+    // Create a 2D array (Chunk) to store integer values on the combined domain
+    ddc::Chunk my_array("my_array", dom, ddc::HostAllocator<int>());
+
+    // Iterate over the first dimension (Dim1)
+    for (ddc::DiscreteElement<Dim1> const idx1 : dom1) {
+        // Iterate over the second dimension (Dim2)
+        for (ddc::DiscreteElement<Dim2> const idx2 : dom2) {
+            // Assign the value 1 to each element in the 2D array
+            my_array(idx1, idx2) = 1;
+
+            // The following would NOT compile as my_array expects a DiscreteElement over Dim2
+            // my_array(idx1, idx1) = 1;
+        }
+    }
+
+    // Extract a 1D view over Dim2 for each idx1
+    for (ddc::DiscreteElement<Dim1> const idx1 : dom1) {
+        // The following would NOT compile if sum_over_dim2 was called
+        // with a `DiscreteDomain<Dim1>`, ensuring type safety.
+        std::cout << sum_over_dim2(my_array[idx1]) << '\n';
+    }
+
+    return 0;
+}
+```
+
+## DDC components
+
+Built on top of DDC core, we also provide optional components as in the Xarray ecosystem. As of today we provide three components: fft, pdi and splines.
+
+### DDC fft and splines
+
+The FFT and splines components both operate on specialized dimensions provided by DDC: `UniformPointSampling` and `NonUniformPointSampling`.
+
+- FFT extension: This component wraps the Kokkos-fft library to provide labelled semantics for the discrete Fourier transform (DFT). It converts data between a uniform sampling and Fourier coefficients in the Fourier basis. Depending on the transformation direction, the input array is defined over `UniformPointSampling`, while the output is defined over `PeriodicPointSampling`, which represents the Fourier dimension in DDC.
+- Splines extension: Similar to FFT, this component provides a spline transform, converting between sampled data and B-spline coefficients in a spline basis. Depending on the transformation direction, the input array is defined over `UniformPointSampling` or `NonUniformPointSampling`, while the output is defined over `UniformBSplines` or `NonUniformBSplines`. (TODO: add a citation)
```

Comment:
What kind of citation do you want here?

Reply:
I had in mind either your thesis manuscript or Yuuichi's paper ?

Reply:
Btw, for citations do you think a github link should only appear as a link in the markdown or should be a reference in the bib ?

Reply:
I usually put links in references but it is up to you. I think Yuuichi's paper should be sufficient. It is specific to this implementation so it makes sense to cite it here I think